weblate/api/views.py

```diff
@@ -1010,6 +1012,41 @@ def file(self, request: Request, **kwargs):
             name=instance.slug,
         )
 
+    @action(detail=True, methods=["get", "post"])
+    def machinery_settings(self, request: Request, **kwargs):
+        """List or create/update machinery configuration for a project."""
+        project = self.get_object()
```

Comment:
GET should be restricted to `project.edit` as well as the configuration can contain credentials for the services.